packages/mui-base/src/useMenuItem/useMenuItemContextStabilizer.ts

```diff
@@ -6,16 +6,16 @@ import { ListContext, ListContextValue, ListItemState } from '../useList';
 /**
  * Stabilizes the ListContext value for the MenuItem component, so it doesn't change when sibling items update.
  *
- * @param id The id of the MenuItem. If undefined, it will be generated with useId.
- * @returns The stable ListContext value and the id of the MenuItem.
- *
  * Demos:
  *
  * - [Menu](https://mui.com/base-ui/react-menu/#hooks)
  *
  * API:
  *
  * - [useMenuItemContextStabilizer API](https://mui.com/base-ui/react-menu/hooks-api/#use-menu-item-context-stabilizer)
+ *
+ * @param id - The id of the MenuItem. If undefined, it will be generated with useId.
+ * @returns The stable ListContext value and the id of the MenuItem.
```

Comment:
The docs API generation is using naming template to find the types to parse. That's why currently have wrong parameter table 🙈

![image](https://github.com/mui/material-ui/assets/45398769/1a0c7bd2-f8b5-4104-b4fd-dd88ac1e9304)


https://github.com/mui/material-ui/blob/bfcb0b66adb21c0f1f0942f9321a54dd83711196/packages/api-docs-builder/ApiBuilders/HookApiBuilder.ts#L547-L549

Reply:
I saw this, yeah! I was about to open an issue about it. It looks like some hooks like `useMenuItemContextStabilizer` and `useOptionContextStabilizer` are missing the `*Parameters` and `*ReturnValue` types. But even if they had those types, it looks like the API generation for the parameter table in hooks only works for hooks that accept a single object parameter. Same for the return value.

We also need to improve how we display the parameters table in the docs since it's not clear that the hook accepts a single object vs separated parameters when looking at it.

<img width="864" alt="image" src="https://github.com/mui/material-ui/assets/7225802/6c274cdd-fc18-4659-8917-46cc61a56ecd">

Reply:
It feels like the script generation should crash in those cases: https://next--material-ui.netlify.app/base-ui/react-menu/hooks-api/#use-menu-item-context-stabilizer. Forcing to either make the API private, or to use the right naming convention for its type.